homeassistant/components/sql/sensor.py

```diff
@@ -391,20 +392,23 @@ def _update(self) -> Any:
                     value = value.isoformat()
                 elif isinstance(value, (bytes, bytearray)):
                     value = f"0x{value.hex()}"
+                extra_state_attributes[key] = value
                 self._attr_extra_state_attributes[key] = value
 
         if data is not None and isinstance(data, (bytes, bytearray)):
             data = f"0x{data.hex()}"
 
         if data is not None and self._template is not None:
-            self._attr_native_value = (
-                self._template.async_render_with_possible_json_value(data, None)
-            )
+            variables = self._template_variables_with_value(data)
+            if self._render_availability_template(variables):
+                self._attr_native_value = self._template.async_render_as_value_template(
+                    self.entity_id, variables, None
+                )
+                self._process_manual_data(variables)
         else:
             self._attr_native_value = data
 
         if data is None:
             _LOGGER.warning("%s returned no results", self._query)
 
         sess.close()
-        return data
```

Comment:
Why is this removed?

Reply:
OK, but then the signature of the method should be updated to show it returns `None`?